really-core/src/main/scala/io/really/gorilla/Replayer.scala

```diff
@@ -0,0 +1,205 @@
+      }
+    case SnapshotFetchError(r, _) =>
+      val msg = "Failed to retrieve a snapshot for that object"
+      shutdownWithObjectSubscriptionAck(msg)
+  }
+
+  def shutdown(reason: String) = {
+    log.warning(s"$replayerId actor is going to die because of: $reason")
+    globals.mediator ! Unsubscribe(r.toString, self)
+    context.stop(self)
+  }
+
+  def shutdownWithObjectSubscriptionAck(reason: String) = {
+    shutdown(reason)
+    objectSubscriber ! SubscriptionFailure(r, 500, reason)
+  }
+
+  def unexpectedMessage(msg: String) = log.warning(s"$replayerId has received an unexpected message: $msg")
+
+  def flushLogEvents(logs: List[EventLog]) = {
+    logs.foreach {
+      log =>
+        self ! log
+    }
+    self ! EndOfEventsLog
+    context.become(serveLogEvents)
+  }
+
+  def waitingCorrectRevision: Receive = _waitingCorrectRevision orElse commonHandler
+
+  def _waitingCorrectRevision: Receive = {
+    case entry: GorillaLogEntry if (entry.rev == min) =>
+      context.become(servePushUpdates)
+    case entry: GorillaLogEntry if (entry.rev > min + 1) =>
+      shutdownWithObjectSubscriptionAck("Inconsistent push updates revision")
+    case entry: GorillaLogEntry =>
+    case entry: PersistentEvent if (entry.event.rev == min) =>
+      context.become(servePushUpdates)
+    case entry: PersistentEvent if (entry.event.rev > min + 1) =>
+      shutdownWithObjectSubscriptionAck("Inconsistent push updates revision")
+    case entry: PersistentEvent =>
+  }
+
+  def serveLogEvents: Receive = _serveLogEvents orElse commonHandler
+
+  def _serveLogEvents: Receive = {
+    case EventLog(event, r, rev, modelVersion, obj, userInfo, ops) =>
+      event match {
+        case "created" =>
+          objectSubscriber ! GorillaLogCreatedEntry(r, obj, rev, modelVersion, userInfo)
+        case "updated" =>
+          objectSubscriber ! GorillaLogUpdatedEntry(r, obj, rev, modelVersion, userInfo,
+            ops.getOrElse(List.empty))
+        case "deleted" =>
+          objectSubscriber ! GorillaLogDeletedEntry(r, rev, modelVersion, userInfo)
+        case _ =>
+      }
+    case EndOfEventsLog =>
+      unstashAll()
+      context.become(servePushUpdates)
+  }
+
+  def servePushUpdates: Receive = _servePushUpdates orElse commonHandler
+
+  def _servePushUpdates: Receive = {
+    case PersistentCreatedEvent(event) =>
+      objectSubscriber ! GorillaLogCreatedEntry(event.r, event.obj, 1L, event.modelVersion, event.context.auth)
+    case PersistentUpdatedEvent(event, obj) =>
+      objectSubscriber ! GorillaLogUpdatedEntry(event.r, obj, event.rev, event.modelVersion, event.context.auth,
+        event.ops)
+    case PersistentDeletedEvent(event) =>
+      objectSubscriber ! GorillaLogDeletedEntry(event.r, event.rev, event.modelVersion, event.context.auth)
+    case SubscriptionManager.Unsubscribe =>
+      shutdownWithObjectSubscriptionAck("Received unsubscribe request")
+  }
+
+  def getEvents(r: R, min: Revision) = {
```

Comment:
What is the returned type of this method?. If it is a unit, you should choose another name may be `replayEvents`